### PR TITLE
Feature/160/general info step for became a tutor

### PR DESCRIPTION
**Pull request by LubomyrKashuba**
created general information step for become a tutor
image

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -6,7 +6,7 @@ sonar.projectName=SpaceToStudy-Client
 # Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
 
 sonar.sources=.
-sonar.exclusions=**/tests/**/*.js, **/src/containers/student-home-page/popular-categories/*.js, **/src/stories/*
+sonar.exclusions=**/tests/**/*.js, **/src/containers/student-home-page/popular-categories/*.js, **/src/stories/*, **/src/constants/**/*.js'
```
Comment (maarekua) [inline sonar-project.properties:9]: If constants have any logic. It should be tested.

File: src/components/become-tutor-steps/firstStep/FirstStep.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { Box, Checkbox, FormControlLabel, Typography } from '@mui/material'
+import { useTranslation } from 'react-i18next'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import img from '~/assets/img/mentor-home-page/become-tutor/first-step.png'
+import { styles } from './first-step.styles'
+
+const FirstStep = ({ data, handleChange, handleBlur, errors }) => {
```
Comment (maarekua) [inline src/components/become-tutor-steps/firstStep/FirstStep.js:7]: FirstStep is not an appropriate name for the component.

Comment (maarekua) [reply src/components/become-tutor-steps/firstStep/FirstStep.js:7]: You should sync with Rostyslav PR with children's buttons.

File: src/constants/validation/become-tutor.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { nameField } from './common'
+
+export const firstName = (value) => {
```
Comment (maarekua) [inline src/constants/validation/become-tutor.js:3]: Why do we need these wrappers ?

File: src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js [hunk 4]
```diff
@@ -1,16 +1,44 @@
...
-const stepLabels = ['General info', 'Languages', 'Study category', 'Experience', 'Documents', 'Photo and Video']
+const BecomeATutor = () => {
+  const stepLabels = ['General info', 'Languages', 'Study category', 'Experience', 'Documents', 'Photo and Video']
 
-const childrenArr = [
-  <Box key="1">1</Box>,
-  <Box key="2">2</Box>
-]
+  const { handleSubmit, handleChange, handleBlur, data, errors } = useForm({
+    initialValues: {
```
Comment (maarekua) [inline src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js:11]: You can move initial values out of the component.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/components/become-tutor-steps/firstStep/FirstStep.js (hunk 2 above)
Comment (LubomyrKashuba) [reply src/components/become-tutor-steps/firstStep/FirstStep.js]: sure

File: src/constants/validation/become-tutor.js (hunk 3 above)
Comment (LubomyrKashuba) [reply src/constants/validation/become-tutor.js]: we don't need these wrappers!)

File: src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js (hunk 4 above)
Comment (LubomyrKashuba) [reply src/containers/mentor-home-page/become-a-tutor/BecomeATutor.js]: done

File: sonar-project.properties (hunk 1 above)
Comment (LubomyrKashuba) [reply sonar-project.properties]: ok

File: src/components/become-tutor-steps/general-info/GeneralInfo.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import { Box, Checkbox, FormControlLabel, Typography } from '@mui/material'
```
Comment (rostyslaw27) [inline src/components/become-tutor-steps/general-info/GeneralInfo.js:1]: rename png file

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/components/become-tutor-steps/general-info/GeneralInfo.js (hunk 5 above)
Comment (LubomyrKashuba) [reply src/components/become-tutor-steps/general-info/GeneralInfo.js:1]: thanks, done

**Review by RestingState — APPROVED** [review]

File: src/components/become-tutor-steps/general-info/GeneralInfo.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import { useTranslation } from 'react-i18next'
+import AppTextField from '~/components/app-text-field/AppTextField'
+import img from '~/assets/img/mentor-home-page/become-tutor/general-info.png'
```
Comment (rostyslaw27) [inline src/components/become-tutor-steps/general-info/GeneralInfo.js:4]: add an empty line before styles

File: src/constants/translations/en/becomeTutor.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "firstStep": {
```
Comment (rostyslaw27) [inline src/constants/translations/en/becomeTutor.json:2]: change to generalInfo

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: src/components/become-tutor-steps/general-info/GeneralInfo.js (hunk 6 above)
Comment (LubomyrKashuba) [reply src/components/become-tutor-steps/general-info/GeneralInfo.js:6]: ok

File: src/constants/translations/en/becomeTutor.json (hunk 7 above)
Comment (LubomyrKashuba) [reply src/constants/translations/en/becomeTutor.json]: done

File: src/components/become-tutor-steps/general-info/GeneralInfo.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
+import { useTranslation } from 'react-i18next'
+import { Box, Checkbox, FormControlLabel, Typography } from '@mui/material'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+
+import img from '~/assets/img/mentor-home-page/become-tutor/general-info.png'
+import { styles } from './general-info.styles'
+
+const GeneralInfo = ({ data, handleChange, handleBlur, errors, btnsBox }) => {
```
Comment (maarekua) [inline src/components/become-tutor-steps/general-info/GeneralInfo.js:9]: It is not a component. It is the container. Move it to the appropriate folder.

**Review by maarekua — CHANGES_REQUESTED** [review]

**Review by tarasSanotskyi — APPROVED** [review]

File: src/components/become-tutor-steps/general-info/GeneralInfo.js (hunk 8 above)
Comment (LubomyrKashuba) [reply src/components/become-tutor-steps/general-info/GeneralInfo.js]: fixed

**Review by rostyslaw27 — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Review by rostyslaw27 — APPROVED** [review]

**Pull request merged**